include/deal.II/fe/fe_enriched.h

```diff
@@ -691,6 +696,311 @@ class FE_Enriched : public FiniteElement<dim, spacedim>
       &output_data) const;
 };
 
+
+
+/**
+ * This namespace consists of a class needed to create a collection
+ * of FE_Enriched finite elements (hp::FECollection)
+ * to be used with hp::DoFHandler
+ * in a domain with multiple, possibly overlapping, sub-domains with individual
+ * enrichment functions.
+ *
+ * To create hp::FECollection, a graph coloring algorithm is used to assign
+ * colors to enrichment functions before creating hp::FECollection. Hence the
+ * name.
+ */
+namespace ColorEnriched
+{
+  /**
+   * An alias template for predicate function which returns a
+   * boolean for a Triangulation<dim,spacedim>::cell_iterator object.
+   *
+   * This is used by helper functions and in the implementation of
+   * ColorEnriched::Helper class.
+   */
+  template <int dim, int spacedim = dim>
+  using predicate_function = std::function<bool(
+    const typename Triangulation<dim, spacedim>::cell_iterator &)>;
+
+  /**
+   * ColorEnriched::Helper class creates a collection of FE_Enriched finite
+   * elements (hp::FECollection) to be used with hp::DoFHandler in a domain
+   * with multiple, possibly overlapping, sub-domains with individual
+   * enrichment functions. Note that the overlapping regions may have
+   * multiple enrichment functions associated with them. This is implemented
+   * using a general constructor of FE_Enriched object which allows different
+   * enrichment functions.
+   *
+   * Consider a domain with multiple enriched sub-domains
+   * which are disjoint i.e. not connected with each other.
+   * To ensure $C^0$ continuity at the interface between
+   * the enriched sub-domain
+   * (characterized by a single enrichment function) and the
+   * non-enriched domain, we can
+   * use an FE_Enriched object in the enriched
+   * sub-domain and in the non-enriched domain
+   * a standard finite element (eg: FE_Q) wrapped into an FE_Enriched
+   * object (which internally uses a dominating FE_Nothing object).
+   * Refer to the documentation on FE_Enriched for more
+   * information on this. It is to be noted that an FE_Enriched
+   * object is constructed using a base FE
+   * (FiniteElement objects) and one or many
+   * enriched FE. FE_Nothing is a dummy enriched FE.
+   *
+   * The situation becomes more
+   * complicated when two enriched sub-domains
+   * share an interface. When the number of enrichment functions are
+   * same for the sub-domains, FE_Enriched object of one sub-domain
+   * is constructed such that each enriched FE is paired (figuratively) with a
+   * FE_Nothing in the FE_Enriched object of the other sub-domain.
+   * For example, let enriched FEs fe_enr1, fe_enr2 correspond
+   * to the two sub-domains. Then the FE_Enriched objects of the two
+   * sub-domains are
+   * FE_Enriched<dim,spacedim>[fe_base, fe_enr1, fe_nothing] and
```

Comment:
> build using \'[fe_base, fe_enr1, fe_nothing]\'

(current text is a bit misleading as it suggests there is such constructor in FE_Enriched)